server/src/main/java/org/opensearch/search/aggregations/metrics/SumIntegralAggregator.java

```diff
@@ -0,0 +1,128 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.search.aggregations.metrics;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.apache.lucene.search.ScoreMode;
+import org.opensearch.common.lease.Releasables;
+import org.opensearch.common.util.BigArrays;
+import org.opensearch.common.util.LongArray;
+import org.opensearch.search.DocValueFormat;
+import org.opensearch.search.aggregations.Aggregator;
+import org.opensearch.search.aggregations.InternalAggregation;
+import org.opensearch.search.aggregations.LeafBucketCollector;
+import org.opensearch.search.aggregations.LeafBucketCollectorBase;
+import org.opensearch.search.aggregations.support.ValuesSource;
+import org.opensearch.search.aggregations.support.ValuesSourceConfig;
+import org.opensearch.search.internal.SearchContext;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Aggregate all docs into a single sum value for integral numbers
+ *
+ * @opensearch.internal
+ */
+public class SumIntegralAggregator extends NumericMetricsAggregator.SingleValue {
+
+    private final ValuesSource.Numeric valuesSource;
+    private final DocValueFormat format;
+    private LongArray sums;
+
+    SumIntegralAggregator(
+        String name,
+        ValuesSourceConfig valuesSourceConfig,
+        SearchContext context,
+        Aggregator parent,
+        Map<String, Object> metadata
+    ) throws IOException {
+        super(name, context, parent, metadata);
+        this.valuesSource = (ValuesSource.Numeric) valuesSourceConfig.getValuesSource();
+        this.format = valuesSourceConfig.format();
+        sums = context.bigArrays().newLongArray(1, true);
+    }
+
+    @Override
+    public ScoreMode scoreMode() {
+        return valuesSource.needsScores() ? ScoreMode.COMPLETE : ScoreMode.COMPLETE_NO_SCORES;
+    }
+
+    @Override
+    public LeafBucketCollector getLeafCollector(LeafReaderContext ctx, final LeafBucketCollector sub) throws IOException {
+        final BigArrays bigArrays = context.bigArrays();
+        final SortedNumericDocValues values = valuesSource.longValues(ctx);
+        return new LeafBucketCollectorBase(sub, values) {
+            @Override
+            public void collect(int doc, long bucket) throws IOException {
+                sums = bigArrays.grow(sums, bucket + 1);
+
+                if (values.advanceExact(doc)) {
+                    final int valuesCount = values.docValueCount();
```

Comment:
@matthewryanwells I think there is an issue with this approach, the sum may overflow quickly. This is also somewhat true for `double`s but that another question

Reply:
@reta that is a fair point, I could probably fix it by using a BigInteger rather than a long, then return the final value as a double

Reply:
The `BigInteger` sounds like a good idea but I think you will have issues passing it through (as `BigInteger`), the only types OpenSearch is able to deal with are `long` and `double` (more or less) :(

Reply:
The bug we are dealing with happens during the math calculations so if we do the calculation using `BigInteger` we can then get it to return the value of a `double`, solving the bug and this overflow problem. The only issue would be how much of a performance hit involving `BigIntegers` would cause, which I can also look into

Reply:
I have updated the `SumIntegralAggregator.java` to use BigInteger in [52b1f3f](https://github.com/opensearch-project/OpenSearch/pull/6984/commits/52b1f3f92921a375dbadaddf59bb36ec1a8d2511) so I believe that this issue has been solved